include/zephyr/drivers/regulator.h

```diff
@@ -159,30 +168,32 @@ struct regulator_common_config {
-		.min_uv = DT_PROP_OR(node_id, regulator_min_microvolt,         \
-				     INT32_MIN),                               \
-		.max_uv = DT_PROP_OR(node_id, regulator_max_microvolt,         \
-				     INT32_MAX),                               \
-		.init_uv = DT_PROP_OR(node_id, regulator_init_microvolt,       \
-				      INT32_MIN),			       \
-		.min_ua = DT_PROP_OR(node_id, regulator_min_microamp,          \
-				     INT32_MIN),                               \
-		.max_ua = DT_PROP_OR(node_id, regulator_max_microamp,          \
-				     INT32_MAX),                               \
-		.startup_delay_us = DT_PROP_OR(node_id, startup_delay_us, 0),  \
-		.off_on_delay_us = DT_PROP_OR(node_id, off_on_delay_us, 0),    \
-		.allowed_modes = (const regulator_mode_t [])                   \
-			DT_PROP_OR(node_id, regulator_allowed_modes, {}),      \
-		.allowed_modes_cnt =                                           \
-			DT_PROP_LEN_OR(node_id, regulator_allowed_modes, 0),   \
-		.initial_mode = DT_PROP_OR(node_id, regulator_initial_mode,    \
-					   REGULATOR_INITIAL_MODE_UNKNOWN),    \
-		.flags = ((DT_PROP_OR(node_id, regulator_always_on, 0U) *      \
-			   REGULATOR_ALWAYS_ON) |                              \
-			  (DT_PROP_OR(node_id, regulator_boot_on, 0U) *        \
-			   REGULATOR_BOOT_ON)),                                \
+#define REGULATOR_DT_COMMON_CONFIG_INIT(node_id)                                     \
+	{                                                                            \
+		.min_uv = DT_PROP_OR(node_id, regulator_min_microvolt,               \
+				     INT32_MIN),                                     \
+		.max_uv = DT_PROP_OR(node_id, regulator_max_microvolt,               \
+				     INT32_MAX),                                     \
+		.init_uv = DT_PROP_OR(node_id, regulator_init_microvolt,             \
+				      INT32_MIN),			             \
+		.min_ua = DT_PROP_OR(node_id, regulator_min_microamp,                \
+				     INT32_MIN),                                     \
+		.max_ua = DT_PROP_OR(node_id, regulator_max_microamp,                \
+				     INT32_MAX),                                     \
+		.startup_delay_us = DT_PROP_OR(node_id, startup_delay_us, 0),        \
+		.off_on_delay_us = DT_PROP_OR(node_id, off_on_delay_us, 0),          \
+		.allowed_modes = (const regulator_mode_t [])                         \
+			DT_PROP_OR(node_id, regulator_allowed_modes, {}),            \
+		.allowed_modes_cnt =                                                 \
+			DT_PROP_LEN_OR(node_id, regulator_allowed_modes, 0),         \
+		.initial_mode = DT_PROP_OR(node_id, regulator_initial_mode,          \
+					   REGULATOR_INITIAL_MODE_UNKNOWN),          \
+		.active_discharge = DT_PROP_OR(node_id, regulator_active_discharge,  \
+					   UINT8_MAX),                               \
+		.flags = ((DT_PROP_OR(node_id, regulator_always_on, 0U) *            \
+			   REGULATOR_ALWAYS_ON) |                                    \
```

Comment:
nit: maybe 2 bits in flags could be used to save some ROM

Reply:
done... sorry for the delay